File: CSharpCodeBase/components/character/damageprocessor.cs
using System;
 
 using System.Collections.Generic;
 
 using System.Linq;
 
 using System.Text;
 
 using UnityEngine;
 
 using System.Collections;
 
 using MainGame.core;
 
 using UnityEngine.EventSystems;
 
 namespace MainGame{
 public class damageprocessor :component{
 var DamageProcessorComponent = Class({__includes=Component, init = public void (self, entity){
       Component.init(self, entity);
       this.entity.damageProcessor = self;
 }})
 public void DamageProcessorComponent:SendDamage(target, damage){
   var damageData = {
     target=target,;
     damage=damage.damage,;
     effects=damage.effects,;
     source=this.entity,;
   }
   target:Message("OnDamageReceive", damageData);
 }
 Component}}

File: CSharpCodeBase/components/character/damageprocessor.lua
local Class = require 'utils.hump.class'
local Component = require 'components.component'

local DamageProcessorComponent = Class({__includes=Component, init = function (self, entity)
      Component.init(self, entity)
      self.entity.damageProcessor = self
end})

function DamageProcessorComponent:SendDamage(target, damage)
  local damageData = {
    target=target,
    damage=damage.damage,
    effects=damage.effects,
    source=self.entity,
  }
  target:Message('OnDamageReceive', damageData)
end

return DamageProcessorComponent